File: packages/crisp/ios/Plugin/Classes/Options/OpenHelpdeskArticleOptions.swift
import Foundation
import Capacitor

@objc public class OpenHelpdeskArticleOptions: NSObject {
    let category: String?
    let id: String
    let locale: String
    let title: String?

    init(_ call: CAPPluginCall) throws {
        guard let id = call.getString("id") else {
            throw CustomError.idMissing
        }
        guard let locale = call.getString("locale") else {
            throw CustomError.localeMissing
        }
        self.id = id
        self.locale = locale
        self.category = call.getString("category")
        self.title = call.getString("title")
    }
}

File: packages/crisp/android/src/main/java/io/capawesome/capacitorjs/plugins/crisp/classes/options/OpenHelpdeskArticleOptions.java
package io.capawesome.capacitorjs.plugins.crisp.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.crisp.classes.CustomExceptions;

public class OpenHelpdeskArticleOptions {

    @Nullable
    private final String category;

    @NonNull
    private final String id;

    @NonNull
    private final String locale;

    @Nullable
    private final String title;

    public OpenHelpdeskArticleOptions(@NonNull PluginCall call) throws Exception {
        String id = call.getString("id");
        if (id == null) {
            throw CustomExceptions.ID_MISSING;
        }
        String locale = call.getString("locale");
        if (locale == null) {
            throw CustomExceptions.LOCALE_MISSING;
        }
        this.id = id;
        this.locale = locale;
        this.category = call.getString("category");
        this.title = call.getString("title");
    }

    @Nullable
    public String getCategory() {
        return category;
    }

    @NonNull
    public String getId() {
        return id;
    }

    @NonNull
    public String getLocale() {
        return locale;
    }

    @Nullable
    public String getTitle() {
        return title;
    }
}
